pkg/apis/garden/types.go

```diff
@@ -328,6 +329,64 @@ const (
 	MachineImageCoreOS MachineImageName = "CoreOS"
 )
 
+////////////////////////////////////////////////////
+//                    PROJECTS                    //
+////////////////////////////////////////////////////
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Project holds certain properties about a Gardener project.
+type Project struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	// +optional
+	metav1.ObjectMeta
+	// Spec defines the project properties.
+	// +optional
+	Spec ProjectSpec
+	// Most recently observed status of the Project.
+	// +optional
+	Status ProjectStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ProjectList is a collection of Projects.
+type ProjectList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta
+	// Items is the list of Projects.
+	Items []Project
+}
+
+// ProjectSpec is the specification of a Project.
+type ProjectSpec struct {
+	// Description is a human-readable description of what the project is used for.
+	// +optional
+	Description *string
+	// Owner is a subject representing a user name, an email address, or any other identifier of a user owning
+	// the project.
+	Owner rbacv1.Subject
```

Comment:
This a single struct, but in the charts above is a list. Which one is the correct one?

Reply:
Both. If the creator has specified an invalid owner or not specified an owner at all then the creator's username is automatically added to the `ClusterRoleBinding` as well. This is to prevent accidental self-exclusion of project creators resulting in ops effort.